### ISSUE TITLE
Calendar not working after docker installation

### ISSUE THREAD

**Issue by fpoulain** [tracim/tracim#2777]
## Bug description & symptom

Calendar refuse to work. Nothing appears in logs.
![Capture d’écran de 2020-03-20 10-25-49](https://user-images.githubusercontent.com/32539071/77151356-0a612a00-6a96-11ea-87b6-63958f2004a8.png)

```
2020-03-20 09:25:23 INFO  [radicale:409][b'uWSGIWorker2Core6'] PROPFIND request for '/agenda/workspace/1/' with depth '0' received from 'xx.xx.xx.xx' (forwarded by 127.0.0.1) using 'Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0'
2020-03-20 09:25:23 INFO  [radicale:471][b'uWSGIWorker2Core6'] Successful login: 'tracim'
2020-03-20 09:25:23 INFO  [radicale:387][b'uWSGIWorker2Core6'] PROPFIND response status for '/agenda/workspace/1/' with depth '0' in 0.047 seconds: 207 Multi-Status
[pid: 618|app: 0|req: 5/9] 127.0.0.1 () {58 vars in 1151 bytes} [Fri Mar 20 09:25:23 2020] PROPFIND /agenda/workspace/1/ => generated 342 bytes in 48 msecs (HTTP/1.1 207) 8 headers in 538 bytes (1 switches on core 6)
2020-03-20 09:25:23 INFO  [radicale:409][b'uWSGIWorker2Core7'] PROPFIND request for '/agenda/workspace/1/' with depth '1' received from 'xx.xx.xx.xx' (forwarded by 127.0.0.1) using 'Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0'
2020-03-20 09:25:23 INFO  [radicale:471][b'uWSGIWorker2Core7'] Successful login: 'tracim'
2020-03-20 09:25:24 INFO  [radicale:387][b'uWSGIWorker2Core7'] PROPFIND response status for '/agenda/workspace/1/' with depth '1' in 1.034 seconds: 207 Multi-Status
[pid: 618|app: 0|req: 6/10] 127.0.0.1 () {58 vars in 1151 bytes} [Fri Mar 20 09:25:23 2020] PROPFIND /agenda/workspace/1/ => generated 623 bytes in 1034 msecs (HTTP/1.1 207) 8 headers in 538 bytes (1 switches on core 7)
2020-03-20 09:27:25 INFO  [radicale:409][b'uWSGIWorker1Core7'] PROPFIND request for '/agenda/workspace/1/' with depth '0' received from 'xx.xx.xx.xx' (forwarded by 127.0.0.1) using 'Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0'
2020-03-20 09:27:25 INFO  [radicale:471][b'uWSGIWorker1Core7'] Successful login: 'tracim'
2020-03-20 09:27:25 INFO  [radicale:387][b'uWSGIWorker1Core7'] PROPFIND response status for '/agenda/workspace/1/' with depth '0' in 0.007 seconds: 207 Multi-Status
[pid: 614|app: 0|req: 5/11] 127.0.0.1 () {58 vars in 1151 bytes} [Fri Mar 20 09:27:25 2020] PROPFIND /agenda/workspace/1/ => generated 342 bytes in 7 msecs (HTTP/1.1 207) 8 headers in 538 bytes (1 switches on core 7)
2020-03-20 09:27:25 INFO  [radicale:409][b'uWSGIWorker2Core0'] PROPFIND request for '/agenda/workspace/1/' with depth '1' received from 'xx.xx.xx.xx' (forwarded by 127.0.0.1) using 'Mozilla/5.0 (X11; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0'
2020-03-20 09:27:25 INFO  [radicale:471][b'uWSGIWorker2Core0'] Successful login: 'tracim'
2020-03-20 09:27:25 INFO  [radicale:387][b'uWSGIWorker2Core0'] PROPFIND response status for '/agenda/workspace/1/' with depth '1' in 0.005 seconds: 207 Multi-Status
[pid: 618|app: 0|req: 7/12] 127.0.0.1 () {58 vars in 1151 bytes} [Fri Mar 20 09:27:25 2020] PROPFIND /agenda/workspace/1/ => generated 623 bytes in 5 msecs (HTTP/1.1 207) 8 headers in 538 bytes (1 switches on core 0)
```

## Diagnostic

unknown

## Expected

Not a bug

## Context and version information

Tracim installed using docker behind a nginx frontend.
Created with:
```
START_WEBDAV=1
START_CALDAV=1
```

and basic sttings in developement.ini :
```
### Main Config for Caldav ##
caldav.enabled = True

### Radicale Proxy config ##
## path to tracim radicale server, usually at localhost:port
caldav.radicale_proxy.base_url = http://localhost:5232

### Radicale config ###
# those params are same as in config file of radicale but syntax
# is different.
# use syntax:
; caldav.radicale.section.param_name = value
# instead of radicale.ini syntax:
; [section]
; param_name = value
caldav.radicale.server.host = localhost:5232
## To make tracim work, we need to disable auth and rights.
caldav.radicale.auth.type = None
caldav.radicale.rights.type = None
## We do not need radicale web ui
caldav.radicale.web.type = None
## for tracim we do use file storage.
caldav.radicale.storage.type = multifilesystem
caldav.radicale.storage.filesystem_folder = %(basic_setup.caldav_storage_dir)s
# CORS config for radicale
caldav.radicale.headers.Access-Control-Allow-Origin = *
caldav.radicale.headers.Access-Control-Allow-Methods =  GET, POST, OPTIONS, PROPFIND, PROPPATCH, REPORT, PUT, MOVE, DELETE, LOCK, UNLOCK
caldav.radicale.headers.Access-Control-Allow-Headers =  User-Agent, Authorization, Content-type, Depth, If-match, If-None-Match, Lock-Token, Timeout, Destination, Overwrite, X-client, X-Requested-With, Prefer
caldav.radicale.headers.Access-Control-Expose-Headers = Etag
```

**Comment by PhilippeAccorsi** [comment]
Hi,

- Which version of tracim do you use?
- What is your member right in this shared space?
- do you have change value of this parameter `basic_setup.website_base_url` in development.ini? If yes, do you have stop and starting tracim again?

**Comment by fpoulain** [comment]
Thank for your help (sorry if this is not a bug, I didn't found something like a forum or ML).

>    Which version of tracim do you use?

Don't know. Got 3 month old algoo/tracim image e9b8163fb382. It display 1.9.1 in /tracim/backend/CHANGES.txt

>   What is your member right in this shared space?

I am root and creator (responsable) of the space.

>    do you have change value of this parameter basic_setup.website_base_url in development.ini? If yes, do you have stop and starting tracim again?

basic_setup.website_base_url = http://localhost:8080

(i remember having changed it and I backtracked; then restarted container).

**Comment by PhilippeAccorsi** [comment]
You need to change `basic_setup.website_base_url = http://localhost:8080` by `basic_setup.website_base_url = http://<domain_or_ip_adresse_availabale_in_front_of_your_nginx>`. if you use ssl in nginx, you can add `https` in website_base_url.

Stopping container and starting Tracim again with docker run command (uwsgi need to be reload in container). Other way is :
`docker exec -it <container_id> /bin/bash` and after that `service uwsgi reload` (failed message in container is normal, uwsgi work good). Now you can `exit` from the container.

You can use docker image `algoo/tracim:2.6.1` or wait today because `algoo/tracim:2.6.2` come this afternoon.

**Comment by fpoulain** [comment]
I did it but i am getting the same error. No more log info. It seems also this break the webdav (displayed) link.

**Comment by fpoulain** [comment]
> It seems also this break the webdav (displayed) link.
No. That's because webdav.base_url can't starts with davs:// (whereas this is the url scheme waited by Nautilus)

**Comment by fpoulain** [comment]
I upgraded to 2.6.2. In js console I am getting those messages:
```
La demande d’accès aux cookies ou au stockage sur « https://undefined/agenda/workspace/3/ » a été bloquée parce que nous bloquons toutes les demandes d’accès au stockage par des sites tiers et que le blocage de contenu est activé.
```

I don't have idea about where does come from undefined.

```
$ grep base_url tracim/etc/development.ini | grep -v '^#\|^;'
basic_setup.website_base_url = https://tracim.psmigrants.org
website.base_url = %(basic_setup.website_base_url)s
webdav.base_url = https://tracim.psmigrants.org
caldav.radicale_proxy.base_url = http://localhost:5232
collaborative_document_edition.collabora.base_url = http://10.21.3.28:9980
```

**Comment by PhilippeAccorsi** [comment]
If you try to access on tracim without nginx, is it working?

We use docker image for tracim 2.6.2 (proxy apache in front of container) and we not have thos kind of error.

Can you give nginx config do you used?

**Comment by fpoulain** [comment]
> Can you give nginx config do you used?

It is a very basic one. Behind classical stuffs (server_name, logs, ...) we get
```
 	location / {
		proxy_pass http://xx.xx.xx.xx:ppppp;
 	}
```
and global proxy params:
```
proxy_set_header Host $host;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;
```

> If you try to access on tracim without nginx, is it working?

Not easy to say. The docker expose a port like 10001 and using ssh tunnels or even socks, I will have hostname and ports problems ; maybe certs issues.

Imho there is a lost value in js frontend ; maybe due to a misconfig. So I get many « undefined » as fqdn in asynchronous call and they fails.
![Capture d’écran de 2020-03-27 15-53-22](https://user-images.githubusercontent.com/32539071/77769039-9bf70b80-7043-11ea-9fcd-2b431ad91f24.png)

**Comment by fpoulain** [comment]
@lebouquetin hum ... btw I didn't encountered problems with webdav ; only with caldav.

**Comment by fpoulain** [comment]
@lebouquetin do you have a specific apache reverse proxy configuration for handling caldav or simply a global ProxyPass?

**Comment by lebouquetin** [comment]
Here comes a working configuration we use:

```
<VirtualHost xxx:443>
    Protocols h2 http/1.1
    ServerName xxx

    RequestHeader unset X-Remote-User
    RequestHeader edit Destination ^https http early

    ProxyPreserveHost On
    ProxyPass / http://127.0.0.1:10011/
    ProxyPassReverse / http://127.0.0.1:10011/
</VirtualHost>
```

(the apache included in the docker image is listening on 10011)

**Comment by fpoulain** [comment]
I tried to attack directly the docker in unsecure http, based on ssh -L (to forward the port from th vm to locally), hosts munging (overriding dns) and socat (to backport the local port from 8080 to 80 with root privileges).
I also changed the website_base_url in developpement.ini.

Unfortunately, some requests are weirdly done using https. I don't know why. I logically it doesn't work (ressources are not loaded). All others features seems to works properly.

I also backported from distant frontend to local the 443 to allow to load https ressources but it is blocked by CORS. I am not sure I will be able to shortcut the nginx on this setup.
![Capture d’écran de 2020-03-28 11-52-13](https://user-images.githubusercontent.com/32539071/77821588-c4d2db80-70eb-11ea-90ab-10df0813cbf0.png)

**Comment by PhilippeAccorsi** [comment]
Can you try to start Tracim from scratch (define other volume), to check if something goes wrong in first installation ?

Also, its not necessary to add START_WEBDAV=1 and START_CALDAV=1 on docker command because this two services are activated by default.

You can add all other settings you need to change directly in docker run comand by adding parameter [visible here](https://github.com/tracim/tracim/blob/develop/backend/doc/setting.md#tracim-22-fully-supported-variables), for exemple `-e TRACIM_WEBSITE__BASE_URL=https://<your_domain> -e TRACIM_WEBDAV__BASE_URL=https://<your_domain>` (webdav is available on linux with `davs` when parameter is like this with `https` ([more info](https://github.com/tracim/tracim/issues/1674) )).

You have also some docker run exemple [here](https://github.com/tracim/tracim/tree/develop/tools_docker#tracim-with-docker).

**Comment by PhilippeAccorsi** [comment]
Try also to clean your browser cache.

**Comment by PhilippeAccorsi** [comment]
@fpoulain do you have solved your problem ?

**Comment by fpoulain** [comment]
No. I will plan to start a new docker from scratch, maybe on my laptop. I am also wondering what to do since @lebouquetin suspect web front-end interference (Nginx). Our front-end servers are not easily shortcut-able and still now we don't plans to have Apache in this rôle.

**Comment by fpoulain** [comment]
Actually I am on a mutualized hosting architecture so I don't think I will change the reverse proxy from nginx to apache. It looks like a dead end from here. Also it is very weird and unbelievable that tracim's caldav doesn't tolerate a nginx proxy.

**Comment by lebouquetin** [comment]
> Also it is very weird and unbelievable that tracim's caldav doesn't tolerate a nginx proxy.

Hi @fpoulain. It is unbelievable because it is not true. Tracim tolerate a nginx proxy.

Our answer is:

1. because of the lack of WebDAV support in NGINX, we decided to focus on fully support Apache. This is the reason why we are able to give you Apache example configuration. In case we would have decided to work with NGINX, then the Webdav support would have been an issue, not allowing to fully support Tracim (which is not in our interest)
2. If you are not interested in using WebDAV you can use NGINX as a reverse proxy. This is not the standard way we suggest to deploy Tracim but it's working.

If you want to use NGINX, you have to "convert" the example apache configuration to nginx equivalent.

The only impossible thing is to use NGINX with WebDAV working correctly. This is the reason why we decided to support Apache instead of NGINX. There is not a technicall issue about CalDAV.

As @PhilippeAccorsi asked you, did you try to access directly to the CalDAV server on 10011 port?

I suggest you:
- [ ] validate the tracim docker image is running on your local computer?
- [ ] validate the tracim docker image is running on your local computer behind an NGINX reverse-proxy?
- [ ] thendeploy on production server according to the reverse-proxy configuration you use in step 2

**Comment by Skylsmoi** [comment]
Closing because of no activity
